README.md

```diff
@@ -1 +1,457 @@
+* [Daniel Holmgren](https://github.com/dholms), [Bluesky](https://blueskyweb.xyz/)
+
+# 0. Abstract
+
+Authorized Wire Authenticated Key Exchange (AWAKE) is an [AKE](https://en.wikipedia.org/wiki/Authenticated_Key_Exchange) built on top of the [UCAN auth token](https://github.com/ucan-wg/spec). AWAKE is similar to other mutual authentication schemes (such as self-signed [mTLS](https://datatracker.ietf.org/doc/html/rfc8705)), but with a focus on authorization and proof. AWAKE leverages the UCAN capability chain to prove access to some resource, validating that the requestor is communicating with a party capable of performing certain actions. This is a helpful root of trust with a well defined context when establishing a secure communications channel.
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+# 1 Introduction
+
+AWAKE bootstraps a secure session on top of a public channel. Key exchanges for point-to-point communication are plentiful, but in open, trusteless protocols, rooting trust can be a barrier for ad hoc communications channels. Two common approaches are to use a trusted certificate authority, or ignore the principal and "merely" establish a point-to-point channel.
+
+Capability-based systems have a helpful philosophy towards a third path. By emphasizing authorization over authentication, they provide a way to know something provable about what the other party "can do", even if they have no sure way of knowing "who they are". One way of phrasing this is that such an agent is "functionally equivalent to the principal in this context". AWAKE makes use of authorization to bootstrap point-to-point sessions that are both secure and mutually trusted.
+
+FIXME add section about KDF in the intro
+
+## 1.1 Payload Fields
+
+All payloads MUST include the "AWAKE Version" field `awv: "0.1.0"`. Payloads MUST also include a message type field `type` (see each stage for the value). All field keys and message type values MUST be case-insensitive.
+
+| Field  | Value          | Description           | Required |
+| ------ | -------------- | --------------------- | -------- |
+| `awv`  | `"0.1.0"`      | AWAKE message version | Yes      |
+| `type` | `awake/<type>` | AWAKE message type    | Yes      |
+
+## 1.2 Roles
+
+| Name      | Role                                                 |
+| --------- | ---------------------------------------------------- |
+| Requestor | The agent opening the session                        |
+| Responder | The agent being contacted by the Requestor           |
+| Attacker  | An attacker attempting to gain access to the channel |
+
+## 2 Sequence
+
+AWAKE proceeds in one connection step, four communication rounds, and an OPTIONAL disconnection:
+
+1. Both parties subscribe to a well-known channel
```

Comment:
Is the channel assumed to be long standing?

If not, how often does the requestor broadcast intent?

If the provider joins _after_ the requestor broadcasts intent, when does it get the update that it needs to re-broadcast?

Reply:
Good questions!

> Is the channel assumed to be long standing?

Yes, it's assumed that it will last at least as long as the handshake

> If not, how often does the requestor broadcast intent?

Whenever they want to start a session (during a session they may exchange a secret to speed up future connections, but not in the protocol)

> If the provider joins after the requestor broadcasts intent, when does it get the update that it needs to re-broadcast?

Yup!